new-components/scanners/gosec/internal/sarif/sarif.go

```diff
@@ -0,0 +1,139 @@
+package sarif
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+)
+
+const (
+	// GoSecOutFormatJson tells the parser to unmarshal the raw data using json umarshaler.
+	GoSecOutFormatJson GoSecOutFormat = "json"
+)
+
+type (
+	// GosecOut represents the sarif output returned by running https://github.com/securego/gosec.
+	GosecOut struct {
```

Comment:
this is not sarif, i think gosec either has a bug or you didn't produce output in the correct format

Reply:
Refactored.